### PR TITLE
started porting from python to rust

### PR DESCRIPTION
**Pull request by lewis-weinberger**
This is in draft mode so you can see where I'm going with things, will submit the full PR at a more developed stage!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,12 @@ venv/
...
 # Ignore VSCode data if present
 .vscode/
+
+# Ignore Emacs and Vim data if present
+*.swp
+*~
+
+# Rust
+rust/fourhills/target/
+rust/fourhill/Cargo.lock
```
Comment (sam-kirby) [inline .gitignore]: Cargo.lock generally shouldn't be excluded when target is a binary.

See https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html

Edit to say I don't know how this applies to webassembly/yew

Comment (lewis-weinberger) [reply .gitignore]: Ah my bad, I must have seen that from a library crate. Fixed in 91fc5e5.

**Comment by lewis-weinberger** [comment]
Trying to port as faithfully as possibly from the original Python, but clippy isn't too happy about some of it. 

e.g. in 91fc5e5 the `StatBlock` has a lot of fields, and hence the`StatBlock::default()` implementation requires a lot of function arguments. 

Once everything is working this should probably be refactored into sub-stat blocks, for example `melee_attacks`, `ranged_attacks` and `multiattack` could be combined... and so on.

**Comment by lewis-weinberger** [comment]
Closing for now until a more mature Rust option for the GUI can be found.

**Pull request closed without merging**